internal/controller/promotions/promotions.go

```diff
@@ -158,205 +168,77 @@ func (r *reconciler) Reconcile(
 		return result, nil
 	}
 
-	newStatus := r.syncPromo(ctx, promo)
+	if promo.Status.Phase == v1alpha1.PromotionPhaseRunning {
+		// anything we've already marked Running, we allow it to continue to reconcile
+	} else if promo.Status.Phase.IsTerminal() {
+		// if promo is already finished, nothing to do
+		return result, nil
+	} else {
+		// promo is Pending. Try to activate it.
+		if !r.pqs.tryActivate(ctx, promo) {
+			// It wasn't our turn. Mark this promo as pending (if it wasn't already)
+			if promo.Status.Phase != v1alpha1.PromotionPhasePending {
+				err = kubeclient.PatchStatus(ctx, r.kargoClient, promo, func(status *kargoapi.PromotionStatus) {
+					status.Phase = v1alpha1.PromotionPhasePending
+				})
+				return result, err
+			}
+			return result, nil
+		}
+	}
 
-	updateErr := kubeclient.PatchStatus(ctx, r.kargoClient, promo, func(status *kargoapi.PromotionStatus) {
-		*status = newStatus
+	logger = logger.WithFields(log.Fields{
+		"stage":   promo.Spec.Stage,
+		"freight": promo.Spec.Freight,
 	})
-	if updateErr != nil {
-		logger.Errorf("error updating Promotion status: %s", updateErr)
+	logger.Debug("executing Promotion")
+
+	// Update promo status as Running to give visibility in UI
```

Comment:
Nit: This isn't only for UI purposes. It's also what allows this active promotion to continue being reconciled, so I would suggest removing the bit that make this look UI-specific:

```suggestion
	// Update promo status as Running
```